lib/response.js

```diff
@@ -872,7 +872,9 @@ res.cookie = function (name, value, options) {
     var maxAge = opts.maxAge - 0
 
     if (!isNaN(maxAge)) {
-      opts.expires = new Date(Date.now() + maxAge)
+      if (opts.expires !== 0) {
+        opts.expires = new Date(Date.now() + maxAge)
+      }
```

Comment:
Allowing `opts.expires` to be of type `number` breaks the consistency, that in all other cases `expires` is of type `Date`. This is also enfored by the TS' type defintion `@types/express` as defined [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L349).

Based on my comment on https://github.com/expressjs/express/issues/5150#issuecomment-1496380734, I would propose to remove ~`opts.expires = new Date(Date.now() + maxAge)`~ entirely. Developers can still set this value as part of the function call. The maintainers will need to decide if this would be a breaking-change.

Reply:
@tjarbo thank you your feedback. I put zero because the docs said:

> expires | Date | Expiry date of the cookie in GMT. If not specified or **set to 0**, creates a session cookie.

Ref https://github.com/expressjs/expressjs.com/blob/867a10d748c4f5c83dfdbc634e210621aa1a34a9/_includes/api/en/4x/res-cookie.md